### PR TITLE
Cypress 2608 staff member tests

### PR DESCRIPTION
**Pull request by karolm-saleor**
I want to merge this change because it adds cypress autotests
#2608 

As a user I should not be able to create staff member with not unique email - SALEOR_3508
As an staff user I should be able to change my password in account settings - SALEOR_3510
As an admin I want to update staff member - SALEOR_3507
As an admin I should not be able to update staff member with not unique email - SALEOR_3509

**PR intended to be tested with API branch:** 

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [x] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: cypress/e2e/staffMembers.js [hunk 1]
```diff
@@ -55,24 +59,28 @@ describe("Staff members", () => {
...
-    fillUpUserDetails(firstName, lastName, emailInvite);
-    getMailActivationLinkForUser(emailInvite).then(urlLink => {
-      cy.clearSessionData().visit(urlLink);
-      fillUpSetPassword(password);
-      expectWelcomeMessageIncludes(`${firstName} ${lastName}`);
-    });
-  });
+  it(
+    "should be able to invite staff user. TC: SALEOR_3501",
+    { tags: ["@staffMembers", "@stagedOnly"] },
```
Comment (szczecha) [inline cypress/e2e/staffMembers.js:64]: Shouldn't we add the tag "@allEnv" for staffMembers tests?  cc: @karola312

File: cypress/e2e/staffMembers.js [hunk 2]
```diff
@@ -163,4 +175,148 @@ describe("Staff members", () => {
...
+    () => {
+      const firstName = faker.name.firstName();
+      const emailInvite = TEST_ADMIN_USER.email;
+      activatePlugin({ id: "mirumee.notifications.admin_email" });
+      cy.visit(urlList.staffMembers)
+        .expectSkeletonIsVisible()
+        .get(STAFF_MEMBERS_LIST.inviteStaffMemberButton)
+        .click({ force: true });
+      fillUpUserDetailsWithNotUniqueEmail(firstName, lastName, emailInvite);
```
Comment (szczecha) [inline cypress/e2e/staffMembers.js:190]: My suggestion is to move the last check `.confirmationErrorMessageShouldAppear();`
from function to test body just to be clear about what was expected in the tests

File: cypress/e2e/staffMembers.js [hunk 3]
```diff
@@ -163,4 +175,148 @@ describe("Staff members", () => {
...
+        .click()
+        .clear()
+        .type(`${TEST_ADMIN_USER.email} {enter}`)
+        .get(BUTTON_SELECTORS.confirm)
+        .click()
+        .confirmationErrorMessageShouldAppear();
+    },
+  );
+
+  // Test blocked by https://github.com/saleor/saleor-dashboard/issues/2847
```
Comment (szczecha) [inline cypress/e2e/staffMembers.js:220]: Should we `xit` or skip this test for now as we don't know when the issue will be fixed?

File: cypress/e2e/staffMembers.js (hunk 2 above)
Comment (karolm-saleor) [reply cypress/e2e/staffMembers.js:190]: ok
(@michalina-graczyk)

File: cypress/e2e/staffMembers.js (hunk 3 above)
Comment (krzysztofzuraw) [reply cypress/e2e/staffMembers.js:220]: I propose `it.skip` - more verbose and way easier to guess what is the intention here even if you don't know `xit` convention.

File: cypress/support/api/requests/Plugins.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ export function activatePlugin({ id, channel, active = true }) {
 
   const mutation = `mutation{
     pluginUpdate(id: "${id}" ${channelLine} input:{
-      active:${active},configuration:
+      active:${active}
```
Comment (krzysztofzuraw) [inline cypress/support/api/requests/Plugins.js:28]: Good catch 👏🏻

File: cypress/e2e/staffMembers.js (hunk 3 above)
Comment (karolm-saleor) [reply cypress/e2e/staffMembers.js:220]: ok
(@michalina-graczyk )

File: cypress/support/pages/userPage.js [hunk 5]
```diff
@@ -36,6 +36,22 @@ export function fillUpUserDetails(firstName, lastName, email) {
     .waitForRequestAndCheckIfNoErrors("@StaffMemberUpdate");
 }
 
+export function fillUpUserDetailsWithNotUniqueEmail(
```
Comment (krzysztofzuraw) [inline cypress/support/pages/userPage.js:39]: **question**: do we have somewhere helper that fill up with unique email?

File: package.json [hunk 6]
```diff
@@ -42,6 +42,7 @@
     "color-convert": "^2.0.1",
     "crc-32": "^1.2.0",
     "currency-codes": "^2.1.0",
+    "cypress-repeat": "^2.3.3",
```
Comment (krzysztofzuraw) [inline package.json:45]: **suggestion**: move this dependency to optional

We should keep `cypress-repeat` and other cypress/e2e deps inside `optionalDependencies` in pkg.json. 

Reference: https://docs.npmjs.com/cli/v9/configuring-npm/package-json#optionaldependencies

Comment (karolm-saleor) [reply package.json:45]: ok

File: cypress/support/pages/userPage.js (hunk 5 above)
Comment (karolm-saleor) [reply cypress/support/pages/userPage.js:39]: Ill change the name as its misleading, good catch

File: cypress/support/pages/userPage.js [hunk 7]
```diff
@@ -15,7 +15,11 @@ export function fillUpSetPassword(password) {
     .waitForRequestAndCheckIfNoErrors("@setPassword");
 }
 
-export function fillUpUserDetails(firstName, lastName, email) {
+export function fillUpUserDetailsAndAddFirstPermission(
```
Comment (karola312) [inline cypress/support/pages/userPage.js:18]: The part of code in this function is the same as in fillUpOnlyUserDetails(). You can do something like

fillUpUserDetailsAndAddFirstPermission(){
    fillUpOnlyUserDetails()

File: cypress/e2e/staffMembers.js [hunk 8]
```diff
@@ -163,4 +175,148 @@ describe("Staff members", () => {
...
     },
   );
+
+  it(
+    "should not be able to create staff member with not unique email. TC: SALEOR_3508",
+    { tags: ["@staffMembers", "@stagedOnly"] },
+    () => {
+      const firstName = faker.name.firstName();
+      const emailInvite = TEST_ADMIN_USER.email;
+      activatePlugin({ id: "mirumee.notifications.admin_email" });
```
[the remark this answers is not in the document]
Comment (karolm-saleor) [reply cypress/e2e/staffMembers.js:185]: ok

File: cypress/e2e/staffMembers.js [hunk 9]
```diff
@@ -163,4 +175,148 @@ describe("Staff members", () => {
...
+      //   .confirmationMessageShouldAppear();
+      // cy.clearSessionData().loginUserViaRequest("auth", {
+      //   email: changedEmail,
+      //   password: Cypress.env("USER_PASSWORD"),
+      // });
+      //
+      // cy.visit(urlList.staffMembers)
+      // expectWelcomeMessageIncludes(
+      //   `${changedName}`,
+      // );
```
[the remark this answers is not in the document]
Comment (karolm-saleor) [reply cypress/e2e/staffMembers.js:273]: ok

File: cypress/support/pages/userPage.js (hunk 7 above)
Comment (karolm-saleor) [reply cypress/support/pages/userPage.js:18]: ok

File: cypress/e2e/staffMembers.js (hunk 1 above)
Comment (karolm-saleor) [reply cypress/e2e/staffMembers.js:64]: ok

**Review by karola312 — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**